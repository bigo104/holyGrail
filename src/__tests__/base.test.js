import React, { useState } from 'react'
// import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

describe('Init rgr for our react stack', () => {
  test('Jest should load and watch dirs', () => {
    const watching = true
    expect(watching).toEqual(true)
  })
})

describe('should render react base dom component', () => {
  afterEach(cleanup)
  const Greeting = () => {
    const [count, countState] = useState(0)
    return (
      <div>
        <button onClick={() => countState(count => count + 1)}>
        increase clicks {count}
        </button>
        <p> shit </p>
      </div>
    )
  }

  test('should render text from dummy html', () => {
    const { getByText } = render(<Greeting/>)
    expect(getByText('shit')).toBeInTheDocument()
  })

  test('should match first snap shot', () => {
    const { getByText } = render(<Greeting/>)
    // const firstRender = asFragment()
    // fireEvent.click(getByText(/increase clicks/))
    const tree = renderer.create(<Greeting/>).toJSON()
    // const { getByText, asFragment } = tree
    fireEvent.click(getByText(/increase clicks/))
    expect(tree).toMatchSnapshot()
  })
})
