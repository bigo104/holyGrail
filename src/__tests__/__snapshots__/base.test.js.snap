// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render react base dom component should match first snap shot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    increase clicks 
    0
  </button>
  <p>
     shit 
  </p>
</div>
`;
